### PR TITLE
test(taiko-client): introduce `taiko-reth` as another L2 node in testing

### LINKED ISSUE #18161: test(taiko-client): introduce `taiko-reth` as another L2 test node

### PR DESCRIPTION
**Pull request by mask-pp**

### CODE REVIEW

File: packages/taiko-client/internal/docker/nodes/docker-compose.yml [hunk 1]
```diff
@@ -16,11 +16,11 @@ services:
...
       - cancun
 
-  l2_execution_engine:
-    container_name: l2_node
+  l2_geth:
+    container_name: l2_geth
     image: us-docker.pkg.dev/evmchain/images/taiko-geth:taiko
     restart: unless-stopped
-    pull_policy: always
+    pull_policy: missing
```
Comment (davidtaikocha) [inline packages/taiko-client/internal/docker/nodes/docker-compose.yml:23]: why not always pulling the latest image?

Comment (mask-pp) [reply packages/taiko-client/internal/docker/nodes/docker-compose.yml:23]: It's an optional change for faster local debugging.

File: packages/taiko-client/driver/chain_syncer/blob/syncer_test.go [hunk 2]
```diff
@@ -111,6 +111,10 @@ func (s *BlobSyncerTestSuite) TestInsertNewHead() {
 }
 
 func (s *BlobSyncerTestSuite) TestTreasuryIncomeAllAnchors() {
+	// TODO: Temporarily skip this test case when use l2_reth node.
+	if os.Getenv("L2_NODE") == "l2_reth" {
```
Comment (davidtaikocha) [inline packages/taiko-client/driver/chain_syncer/blob/syncer_test.go:115]: Let's use `L2_RETH_NODE`  to make it more clear?

File: packages/taiko-client/driver/chain_syncer/blob/syncer_test.go [hunk 3]
```diff
@@ -133,6 +137,10 @@ func (s *BlobSyncerTestSuite) TestTreasuryIncomeAllAnchors() {
 }
 
 func (s *BlobSyncerTestSuite) TestTreasuryIncome() {
+	// TODO: Temporarily skip this test case when use l2_reth node.
```
Comment (davidtaikocha) [inline packages/taiko-client/driver/chain_syncer/blob/syncer_test.go:140]:
```suggestion
	// TODO: Temporarily skip this test case when using l2_reth node.
```

**Review by YoGhurt111 — APPROVED** [review]

File: packages/taiko-client/driver/chain_syncer/blob/syncer_test.go (hunk 2 above)
Comment (mask-pp) [reply packages/taiko-client/driver/chain_syncer/blob/syncer_test.go:115]: `L2_NODE` is a common name and the value may be l2_reth or l2_geth.

File: packages/taiko-client/internal/docker/nodes/docker-compose.yml [hunk 4]
```diff
@@ -4,7 +4,7 @@ services:
     image: ghcr.io/foundry-rs/foundry:nightly
     restart: unless-stopped
     platform: linux/amd64
-    pull_policy: always
+    pull_policy: missing
```
[the remark this answers is not in the document]
Comment (mask-pp) [reply packages/taiko-client/internal/docker/nodes/docker-compose.yml]: reverted.

File: packages/taiko-client/proposer/proposer_test.go [hunk 5]
```diff
@@ -171,6 +171,10 @@ func (s *ProposerTestSuite) TestProposeTxLists() {
 }
 
 func (s *ProposerTestSuite) TestProposeOpNoEmptyBlock() {
+	// TODO: Temporarily skip this test case when use l2_reth node.
```
Comment (davidtaikocha) [inline packages/taiko-client/proposer/proposer_test.go:174]:
```suggestion
	// TODO: Temporarily skip this test case when using l2_reth node.
```

File: packages/taiko-client/driver/chain_syncer/blob/syncer_test.go [hunk 6]
```diff
@@ -111,6 +111,10 @@ func (s *BlobSyncerTestSuite) TestInsertNewHead() {
 }
 
 func (s *BlobSyncerTestSuite) TestTreasuryIncomeAllAnchors() {
+	// TODO: Temporarily skip this test case when use l2_reth node.
```
Comment (davidtaikocha) [inline packages/taiko-client/driver/chain_syncer/blob/syncer_test.go:114]:
```suggestion
	// TODO: Temporarily skip this test case when using l2_reth node.
```

**Review by davidtaikocha — APPROVED** [review]

File: .github/workflows/taiko-client--test.yml [hunk 7]
```diff
@@ -55,8 +59,10 @@ jobs:
       - name: Install pnpm dependencies
         uses: ./.github/actions/install-pnpm-dependencies
 
-      - name: Test
+      - name: Run Tests on ${{ matrix.execution_node }} execution node
```
Comment (davidtaikocha) [inline .github/workflows/taiko-client--test.yml:62]:
```suggestion
      - name: Run Tests on ${{ matrix.execution_node }} execution engine```

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**